extensions/publisher-cves/src/components/VulnerabilityChecker.vue

```diff
@@ -0,0 +1,272 @@
+          const fixedVersion = getFixedVersion(vuln);
+
+          if (!packageMap.has(packageId)) {
+            packageMap.set(packageId, {
+              packageInfo: pkg,
+              vulnerabilities: [],
+              repo,
+              fixedVersions: [],
+            });
+          }
+
+          const packageData = packageMap.get(packageId)!;
+          packageData.vulnerabilities.push(vuln);
+
+          if (fixedVersion) {
+            packageData.fixedVersions.push(fixedVersion);
+          }
+        }
+      }
+    }
+  }
+
+  // Convert the Map to an array and determine the latest fixed version
+  return Array.from(packageMap.values()).map((item) => {
+    // Sort fixed versions semantically (assuming they are valid semver)
+    // This simple comparison works for most simple version formats
+    const sortedFixedVersions = [...item.fixedVersions].sort((a, b) => {
+      const aParts = a.split(".");
+      const bParts = b.split(".");
+
+      for (let i = 0; i < Math.max(aParts.length, bParts.length); i++) {
+        const aNum = parseInt(aParts[i] || "0", 10);
+        const bNum = parseInt(bParts[i] || "0", 10);
+        if (aNum !== bNum) {
+          return bNum - aNum; // Descending order (latest first)
+        }
+      }
+
+      return 0;
+    });
+
+    return {
+      packageInfo: item.packageInfo,
+      vulnerabilities: item.vulnerabilities,
+      repo: item.repo,
+      latestFixedVersion:
+        sortedFixedVersions.length > 0 ? sortedFixedVersions[0] : null,
+    };
+  });
+});
+
+// Status messages based on loading state
+const loadingMessage = computed(() => {
+  if (isLoadingPackages.value && isLoadingVulns.value) {
+    return "Loading package and vulnerability data...";
+  } else if (isLoadingPackages.value) {
+    return "Loading package data...";
+  } else if (isLoadingVulns.value) {
+    return "Loading vulnerability data...";
+  }
+  return "";
+});
+
+// Get the current content information and derived properties
+const contentInfo = computed(() => contentStore.currentContent);
+const contentTitle = computed(
+  () => contentInfo.value?.title || "Unnamed Content",
+);
+const dashboardUrl = computed(() => contentInfo.value?.dashboard_url || null);
+
+// Stats
+const totalPackages = computed(() => packages.value.length);
+const pythonPackages = computed(
+  () =>
+    packages.value.filter((p) => p.language.toLowerCase() === "python").length,
+);
+const rPackages = computed(
+  () => packages.value.filter((p) => p.language.toLowerCase() === "r").length,
+);
+
+// Total number of vulnerabilities (CVEs) across all packages
+const totalVulnerabilities = computed(() => {
+  return vulnerablePackages.value.reduce((total, pkg) => {
+    return total + pkg.vulnerabilities.length;
+  }, 0);
+});
+</script>
+
+<template>
+  <!-- Only show the component if we have content to analyze -->
+  <div
+    v-if="hasPackages || isLoadingPackages"
+    class="max-w-4xl mx-auto p-5 bg-white rounded-lg shadow-md"
+  >
```

Comment:
Perhaps this is why things without packages don't show anything when you click on them 🤔

Reply:
Definitely part of it. I'll have to add some fallback behavior if there aren't any packages.